source/Mission.cpp

```diff
@@ -151,6 +151,20 @@ void Mission::Load(const DataNode &node)
 			blocked = child.Token(1);
 		else if(child.Token(0) == "deadline" && child.Size() >= 4)
 			deadline = Date(child.Value(1), child.Value(2), child.Value(3));
+		else if(child.Size() == 2 && child.Token(0) == "complete" && child.Token(1) == "anywhere")
+			completeAnywhere = true;
+		else if(child.Size() == 2 && child.Token(0) == "complete" && child.Token(1) == "at")
+		{
+			LocationFilter loaded(child);
+			if(!loaded.IsValid())
+				continue;
+			if(loaded.IsEmpty())
+				child.PrintTrace("Error: The \"complete at\" filter must not be empty. Ignoring this line.");
+			else
+				completionFilter = loaded;
+		}
+		else if(child.Token(0) == "complete")
+			child.PrintTrace("Invalid \"complete\" option.");
```

Comment:
These should probably go near where the source and destination are loaded (I'd put it immediately after destination.)

Also, I know comparing the size is faster, but that performance difference isn't really important when loading stuff and everything else puts the strings first.
Do:
```c++
else if(child.Token(0) == "complete" && child.Size() == 2)
{
	if(child.Token(1) ==  "anywhere")
		// Whatever
	else if(child.Token(1) == "at")
		// Whatever
	else
		// Print trace
}
```

Reply:
I moved it after destination, and did something similar to this.